Added resource demand linked entities

CsarRamos opened the pull request:
Implemented logic in the resource demands class to add the referenced entities.

Tested with the event class.

**Before:**

In Searchkit  it's not possible to join the event class to the resource demand class.

**After:**

Searchkit allows join event class to the resource demands class.

https://github.com/systopia/de.systopia.resource/issues/27

bjendres commented on CRM/Resource/DAO/ResourceDemand.php:
```diff
@@ -113,6 +113,17 @@ public static function getReferenceColumns() {
   }
 
+  /**
+   * CAREFUL: DON'T DELETE THIS WHEN REBUILDING THE DAOs! */
```
Great work, @CsarRamos, but shouldn't there be a way to define this sort of thing in [the schema](https://github.com/systopia/de.systopia.resource/blob/master/xml/schema/CRM/Resource/ResourceDemand.xml) so we wouldn't have to worry about the DAO rebuild? @colemanw, would you happen to know this off the top of your head?

colemanw replied:
@bjendres yes - `buildOptions` funtions belong in the BAO file, not the DAO. Just move it there and all should be good.

bjendres replied:
Thanks! That's exactly what I wanted to hear :-)

CsarRamos replied:
Thanks for the help, I tried it but it didn't work, the **$fieldName** in `buildOptions` return `resource_type_id` instead of `entity_table` when compose join in searchkit. I tried to replicate the same logic on Resource class: https://github.com/systopia/de.systopia.resource/blob/master/CRM/Resource/DAO/Resource.php#L99
maybe I need to rethink the approach and modify schema. Any suggestions are welcome, thanks!

colemanw commented:
Ok I've read this PR more carefully and I think I understand what you're trying to do. I recommend:

1. Don't use the `grouping` column to store the table name, use `value` - that's how it's done elsewhere.
2. You don't need a `buildOptions` function at all. Just add `<pseudoconstant>` tag pointing to your option group like this: https://github.com/civicrm/civicrm-core/blob/a6d0f90f8da1d21cd87e62f56d9bf4dc8f05d1ef/xml/schema/Core/EntityTag.xml#L26-L33
3. After modifying xml you need to run `civix generate:entity-boilerplate`.
4. Also, I think it would be better to package your new option group in a `.mgd.php` file instead of `.json` as that's a newer approach and we ought to be removing the `.json` stuff from this extension.

bjendres commented:
Thanks for helping out, @colemanw. 

@CsarRamos let us know if you get stuck.

CsarRamos commented:
> Thanks for helping out, @colemanw.
> 
> @CsarRamos let us know if you get stuck.

Great, thank you for te guidelines, I'll let you if there's any problem, thanks.

CsarRamos commented:
> Hello @bjendres I have added some comits that implement the functionality following @colemanw recommendations. I have tried to modify the resource class, but it seems that some decisions have to be made on how to implement it and I am quite unsure how to do it.

Apologies, I updated civix and did not detect that there were errors, I will check it carefully and update it as soon as possible, thanks.

bjendres commented:
@jensschuppe, can we review this together some time soon?

colemanw commented on resources/OptionGroup_resource_demand_types.mgd.php:
```diff
@@ -0,0 +1,54 @@
+          'label',
+          'description',
+        ],
+      ],
```
Safeguard to prevent errors if the group already exists:
```suggestion
      ],
      'match' => ['name'],
```

CsarRamos commented:
Hello, I added new change to allow to add the label field in the form build with afform. 
If it should be a new requirement, please let me know and I will create another issue/PR. Thanks!

bjendres commented:
@jensschuppe Could you have a look at this?
